napari/layers/shapes/_shapes_utils.py

```diff
@@ -979,6 +981,35 @@ def points_in_poly(points, vertices):
     return inside
 
 
+def extract_shape_type(data, shape_type=None):
+    """Checks whether data contains shape type information and extracts it if so, returning
+    data and shape_type separately.
+
+    Parameters
+    ----------
+    data : Array | Tuple(Array,str) | List[Array | Tuple(Array, str)] | Tuple(List[Array], str)
+        list or array of vertices belonging to each shape, optionally containing shape type strings
+    shape_type : str | None
+        metadata shape type string, or None if none was passed
```

Comment:
Now that I have real comments below, I'm ok with asking this to be fixed too :joy: Can you wrap at 80c, and top line should be a one-liner?  e.g. "Return data and shape type separately if shape type is provided." You can expand on this in a paragraph under that if needed.

@Carreau I'm ready for us to deploy your docstring validator :joy:

Reply:
Didn't realize black doesn't format docstrings to wrap at 80...